### ISSUE TITLE
Error running Farthest On Circle tool

### ISSUE THREAD

**Issue by JudTown17** [Esri/solutions-geoprocessing-toolbox#404]
@mfunk , @jfrygeo :

The following error occurs when running the Farthest On Circle tool in the Maritime Decision Aid Tools_10.4.tbx toolbox. The error happens in ArcMap 10.4 and also running it in Python.

__INFO: 2016-03-21 11:32:18,956 Running tool (Farthest On Circle)
ERROR: 2016-03-21 11:32:21,865 Executing: FarthestOnCircle "C:\Data\Defense Solutions\solutions-geoprocessing-toolbox\data\data.gdb\Ships" 150 15 "C:\Data\Defense Solutions\solutions-geoprocessing-toolbox\data\output.gdb\hoursOutput"
Start Time: Mon Mar 21 11:32:19 2016
Executing (Calculate Value): CalculateValue Convert_Knots_to_MetersPerHour("15") "def Convert_Knots_to_MetersPerHour(knots):\n  # convert knots to hours per meter\n  # to use as cost surface value (cost to cross 1 meter)\n\n  # convert knots to km per hour\n  # 1 knot = 1 nautical mile per hour\n  # 1 nautical mile = 1852 meters\n  # 1 knot = 1.852 kmph, 1 knot = 1852 meters per hour\n\n  dblKnots = float(knots)\n  MetersPerHr = dblKnots * 1852 \n  HrsPerMeter = 1 / MetersPerHr \n  return HrsPerMeter\n" Double
Start Time: Mon Mar 21 11:32:20 2016
Value = 3.59971202303816E-05
Succeeded at Mon Mar 21 11:32:20 2016 (Elapsed Time: 0.09 seconds)
Executing (Calculate Value (2)): CalculateValue Convert_NauticalMiles_to_Meters("150") "def Convert_NauticalMiles_to_Meters(dist_nm):\n  # convert nautical miles to meters\n  # to use as cost surface value\n\n  # convert knots to km per hour\n  # 1 nautical mile = 1852 meters\n\n  dblNauticalMiles = float(dist_nm)\n  dblMeters = dblNauticalMiles * 1852 \n  return dblMeters \n" Double
Start Time: Mon Mar 21 11:32:20 2016
Value = 277800
Succeeded at Mon Mar 21 11:32:20 2016 (Elapsed Time: 0.09 seconds)
Executing (Buffer): Buffer "C:\Data\Defense Solutions\solutions-geoprocessing-toolbox\data\data.gdb\Ships" in_memory\lastseen_Buffer 277800 FULL ROUND NONE # PLANAR
Start Time: Mon Mar 21 11:32:20 2016
Succeeded at Mon Mar 21 11:32:20 2016 (Elapsed Time: 0.08 seconds)
Executing (Raster Calculator): RasterCalculator Con("C:\Data\Defense Solutions\solutions-geoprocessing-toolbox\data\data.gdb\watermaskworld1km", 3.59971202303816E-05) in_memory\SpeedCostSurf
Start Time: Mon Mar 21 11:32:20 2016
Con(Raster(r"C:\Data\Defense Solutions\solutions-geoprocessing-toolbox\data\data.gdb\watermaskworld1km"), 3.59971202303816E-05)
ERROR 000539: Error running expression: rcexec() 
Traceback (most recent call last):
  File "<expression>", line 1, in <module>
  File "<string>", line 8, in rcexec
RuntimeError: ERROR 010240: Could not save raster dataset to in_memory\SpeedCostSurf with output format MEM.

Failed to execute (Raster Calculator).
Failed at Mon Mar 21 11:32:21 2016 (Elapsed Time: 0.63 seconds)
Failed to execute (FarthestOnCircle).
Failed at Mon Mar 21 11:32:21 2016 (Elapsed Time: 2.18 seconds)__

**Comment by jfrygeo** [comment]
I am running 10.4 as well and I am able to run the Farthest On Circle tool with in_memory\SpeedCostSurf.

**Comment by jfrygeo** [comment]
@JudTown17 try running the Maritime Decision Aid Tools_10.4.tbx toolbox - Farthest On Circle in ArcMap 10.4 by dropping a location with the feature set. I believe you get the:

<i> RuntimeError: ERROR 010240: Could not save raster dataset to in_memory\SpeedCostSurf with output format MEM.</i> 

when it receives a blank position.

**Comment by JudTown17** [comment]
@jfrygeo In ArcMap 10.4 I ran Farthest On Circle with dropping a location for Position Last Seen and it worked. Then I tried it again with a blank position, and I got the same error that I reported in the original post.

**Comment by jfrygeo** [comment]
Yeah, that is as expected. You need to specify a position. The tool will error if no position is set as an input.


Sent from my iPhone

On Apr 1, 2016, at 5:12 PM, Judson Hardy <notifications@github.com<mailto:notifications@github.com>> wrote:


@jfrygeo<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_jfrygeo&d=CwMCaQ&c=n6-cguzQvX_tUIrZOS_4Og&r=LOFtvHGMxkJ_DWps2cZi-A&m=6bUGxIFESJp08gsMh2bDpa7h9HCdPSI2srAjU3hSrPk&s=G5IZxd9Mm4668vj7tavtUbhEjvOf24vnGLcXxA69iZ0&e=> In ArcMap 10.4 I ran Farthest On Circle with dropping a location for Position Last Seen and it worked. Then I tried it again with a blank position, and I got the same error that I reported in the original post.

-

**Comment by jfrygeo** [comment]
@JudTown17 I will make the input position a required option in the model.

**Comment by jfrygeo** [comment]
@JudTown17 The input (a feature set) is required, however it still allows for the user to not specify a position in the feature set, meaning the user could specify a null location and cause the tool to error. The user must specify (a) location(s) for the tool to run.

**Comment by jfrygeo** [comment]
When specifying an input location for the Farthest on Circle Tool:
<li> The tool at ..\solutions-geoprocessing-toolbox\suitability\toolboxes\Maritime Decision Aid Tools_10.4.tbx

<i> runs in ArcMap 10.4 in about 7 Seconds</i>

<li> The tool was created for ArcGIS Pro at ..\solutions-geoprocessing-toolbox\suitability\toolboxes\Maritime Decision Aid Tools.tbx. This tool is named "Farthest on Circle - Original."

<i> runs in ArcGIS Pro in about <b> 16 Minutes</b>.
<i> I tried using the same methodology in the ArcMap version of the tool (by using extent of the range buffer) to be the processing extent for the cost distance tools, but looks like the processing extent environment variable isn't honored, hence 16 minutes of processing.

<li> The tool was also created for ArcGIS Pro at ..\solutions-geoprocessing-toolbox\suitability\toolboxes\Maritime Decision Aid Tools.tbx, but used a different methodology. This tool is named "Farthest on Circle." 

<i> runs in ArcGIS Pro in about <b> 1 Minute</b>. I used a "Clip" tool to process a mask and pass these variables to other tools. Runs much quicker than before in ArcGIS Pro, but much much slower than ArcMap. 

@mfunk @JudTown17 The updated tools are in the most recent pull request.
